addon/index.js

```diff
@@ -71,13 +72,26 @@ function updatePrefs() {
   Object.assign(settings, {
     BASE_URL: env.BASE_URL,
     ALLOWED_ORIGINS: env.BASE_URL + '/*',
+    ALLOWED_ORIGINS_READONLY: 'https://www.mozilla.org/*,about:home',
     HOSTNAME: URL(env.BASE_URL).hostname, // eslint-disable-line new-cap
     TESTPILOT_PREFIX: env.TESTPILOT_PREFIX
   });
 
-  // Set up new PageMod, destroying any previously existing one.
-  if (currentPageMod) { currentPageMod.destroy(); }
-  currentPageMod = new PageMod({
+  // Destroy previously existing PageMods
+  if (readPageMod) { readPageMod.destroy(); }
+  if (writePageMod) { writePageMod.destroy(); }
+
+  // Set up new PageMod for read access to detect Test Pilot installation.
+  readPageMod = new PageMod({
+    include: settings.ALLOWED_ORIGINS_READONLY.split(','),
+    contentScriptFile: './message-bridge-readonly.js',
+    contentScriptWhen: 'start',
+    attachTo: ['top', 'existing'],
+    onAttach: setupApp
```

Comment:
This shouldn't be called here, since `setupApp` is called in `onAttach` for the new `writePageMod` too.